sonar-scm-git-plugin/src/main/java/org/sonarsource/scm/git/JGitBlameCommand.java

```diff
@@ -54,41 +52,23 @@ public JGitBlameCommand(PathResolver pathResolver) {
   @Override
   public void blame(BlameInput input, BlameOutput output) {
     File basedir = input.fileSystem().baseDir();
-    Repository repo = buildRepository(basedir);
-    try {
-      Git git = Git.wrap(repo);
+    try (Repository repo = buildRepository(basedir); Git git = Git.wrap(repo)) {
       File gitBaseDir = repo.getWorkTree();
-      ExecutorService executorService = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
-      List<Future<Void>> tasks = submitTasks(input, output, git, gitBaseDir, executorService);
-      executorService.shutdown();
-      waitForTaskToComplete(tasks);
-    } finally {
-      repo.close();
-    }
-  }
-
-  static void waitForTaskToComplete(List<Future<Void>> tasks) {
-    for (Future<Void> task : tasks) {
+      Stream<InputFile> stream = StreamSupport.stream(input.filesToBlame().spliterator(), true);
+      ForkJoinPool forkJoinPool = new ForkJoinPool(Runtime.getRuntime().availableProcessors(),
+        new GitThreadFactory(), JGitBlameCommand::uncaughtException, false);
+      forkJoinPool.submit(() -> stream.forEach(inputFile -> blame(output, git, gitBaseDir, inputFile)));
       try {
-        task.get();
-      } catch (ExecutionException e) {
-        // Unwrap ExecutionException
-        throw e.getCause() instanceof RuntimeException ? (RuntimeException) e.getCause() : new IllegalStateException(e.getCause());
+        forkJoinPool.shutdown();
+        forkJoinPool.awaitTermination(Long.MAX_VALUE, TimeUnit.SECONDS);
       } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2142)